docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -268,9 +268,157 @@ kube-dns-{network} shall be managed by Endpoints controller just like a regular
+
+Network-isolation mode makes more sense in most cases. Below is an example for specific network, my-network, in one namespace. Similar network policies need to apply to other namespaces, too.
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: network-isolated-policy-for-my-network
+  tenant: foo
+  namesapce: bar
+spec:
+  podSelector:
+    matchLabels:
+      arktos.futurewei.com/network: my-network
+  policyTypes:
+  - Ingress
+  - Egress
+  ingress:
+  - from:
+    - namespaceSelector: {}
+    - podSelector:
+        matchLabels:
+          arktos.futurewei.com/network: my-network
+  egress:
+  - to:
+    - namespaceSelector: {}
+    - podSelector:
+        matchLabels:
+          arktos.futurewei.com/network: my-network
+```
+
+For the default network, isolation can be achieved by:
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: implicit-default-network-isolated-policy
+  tenant: foo
+  namesapce: bar
+spec:
+  podSelector:
+    matchExpressions:
+    - key: arktos.futurewei.com/network
+      operator: DoesNotExist
+  policyTypes:
+  - Ingress
+  - Egress
+  ingress:
+  - from:
+    - namespaceSelector: {}
+    - podSelector:
+        matchLabels:
+          arktos.futurewei.com/network: default
+  - from:
+    - namespaceSelector: {}
+    - podSelector:
+        matchExpressions:
+        - key: arktos.futurewei.com/network
+        operator: DoesNotExist
+  egress:
+  - to:
+    - namespaceSelector: {}
+    - podSelector:
+        matchLabels:
+          arktos.futurewei.com/network: my-network
+  - to:
+    - namespaceSelector: {}
+    - podSelector:
+        matchExpressions:
+        - key: arktos.futurewei.com/network
+        operator: DoesNotExist
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: explcit-default-network-isolated-policy
+  tenant: foo
+  namesapce: bar
+spec:
+  podSelector:
+    matchLabels:
+      arktos.futurewei.com/network: default
+  policyTypes:
+  - Ingress
+  - Egress
+  ingress:
+  - from:
+    - namespaceSelector: {}
+    - podSelector:
+        matchLabels:
+          arktos.futurewei.com/network: default
+  - from:
+    - namespaceSelector: {}
+    - podSelector:
+        matchExpressions:
+        - key: arktos.futurewei.com/network
+        operator: DoesNotExist
+  egress:
+  - to:
+    - namespaceSelector: {}
+    - podSelector:
+        matchLabels:
+          arktos.futurewei.com/network: my-network
+  - to:
+    - namespaceSelector: {}
+    - podSelector:
+        matchExpressions:
+        - key: arktos.futurewei.com/network
+        operator: DoesNotExist
+```
+
+Leveraging existent k8s network policy syntax in Arktos system (which is multi-tenant & has network as new resource type), is quite wordy. If reusing existing k8s network policy enforcer is not a goal, new network policy type as depicted below seems more expressive, at cost of implementing new policy enforcers.
```

Comment:
I suggest that we first make the existing network policy syntax working in a tenant space, just like what we did to other policy objects like roles and clusterRoles.  So that it provides backward compatibility and tenant users can still use their familiar tools.

Reply:
agreed. will update doc to make this point clear.